libs/librepcb/projecteditor/boardeditor/fsm/bes_addvia.cpp

```diff
@@ -368,31 +485,116 @@ void BES_AddVia::sizeEditValueChanged(const PositiveLength& value) noexcept {
   }
 }
 
-void BES_AddVia::drillDiameterEditValueChanged(
-    const PositiveLength& value) noexcept {
+void BES_AddVia::drillDiameterEditValueChanged(const PositiveLength& value)
+      noexcept {
   mCurrentViaDrillDiameter = value;
   if (mViaEditCmd) {
     mViaEditCmd->setDrillDiameter(mCurrentViaDrillDiameter, true);
   }
 }
 
 void BES_AddVia::setNetSignal(NetSignal* netsignal) noexcept {
+  Q_ASSERT(mSubState == SubState_PositioningVia);
   try {
-    if (!netsignal) throw LogicError(__FILE__, __LINE__);
-    mCurrentViaNetSignal = netsignal;
+    if (!netsignal) {
+      throw LogicError(__FILE__, __LINE__);
+    }
```

Comment:
During testing, sometimes the via position was not updated at all when moving the cursor, which seemed to be very strange since `setPosition()` was called properly. After some debugging, it turned out that the `setNetSignal()` is now called every time in the MouseMove event, which constantly removes the whole net segment and re-adds it immediately, which seems to lead to graphical issues (in my case, it was just never repainted, although I'd expect flickering...). Updating the net signal only when needed, fixed that issue:

```suggestion
    }
    if (netsignal == &mCurrentVia->getNetSignalOfNetSegment()) {
      return;  // net signal is already correct, no update needed
    }
```